lisa/src/main/java/it/unive/lisa/analysis/heap/PointBasedHeap.java

```diff
@@ -0,0 +1,247 @@
+package it.unive.lisa.analysis.heap;
+
+import it.unive.lisa.analysis.BaseLattice;
+import it.unive.lisa.analysis.HeapDomain;
+import it.unive.lisa.analysis.SemanticException;
+import it.unive.lisa.program.cfg.ProgramPoint;
+import it.unive.lisa.symbolic.SymbolicExpression;
+import it.unive.lisa.symbolic.heap.AccessChild;
+import it.unive.lisa.symbolic.heap.HeapAllocation;
+import it.unive.lisa.symbolic.heap.HeapReference;
+import it.unive.lisa.symbolic.value.BinaryExpression;
+import it.unive.lisa.symbolic.value.HeapIdentifier;
+import it.unive.lisa.symbolic.value.Identifier;
+import it.unive.lisa.symbolic.value.Skip;
+import it.unive.lisa.symbolic.value.UnaryExpression;
+import it.unive.lisa.symbolic.value.ValueExpression;
+import it.unive.lisa.symbolic.value.ValueIdentifier;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.collections.CollectionUtils;
+
+/**
+ * A point-based heap implementation that abstracts heap locations depending on
+ * their allocation sites, namely the position of the code wehre heap locations
+ * are generated. All heap locations that are generated at the same allocation
+ * sites are abstracted into a single unique identifier.
```

Comment:
Check if there is a reference paper for this implementation, and add the citation information if possible